sbi/utils/sbiutils.py

```diff
@@ -884,6 +885,61 @@ def within_support(distribution: Any, samples: Tensor) -> Tensor:
         return torch.isfinite(distribution.log_prob(samples))
 
 
+def mcmc_transform(
+    prior: Any,
+    num_prior_samples_for_zscoring: int = 1000,
+    enable_transform: bool = True,
+    device: str = "cpu",
+    **kwargs,
+) -> torch_tf.Transform:
+    """
+    Builds a transform that is applied to parameters during MCMC.
+
+    It does two things:
+    1) When the prior support is bounded, it transforms the parameters into unbounded
+        space.
+    2) It z-scores the parameters such that MCMC is performed in a z-scored space.
+
+    Args:
+        prior: The prior distribution.
+        num_prior_samples_for_zscoring: The number of samples drawn from the prior
+            to infer the `mean` and `stddev` of the prior used for z-scoring. Unused if
+            the prior has bounded support or when the prior has `mean` and `stddev`
+            attributes.
+        enable_transform: Whether or not to use a transformation during MCMC.
+
+    Returns: A transformation that transforms whose `forward()` maps from unconstrained
+        (or z-scored) to constrained (or non-z-scored) space.
+    """
+
+    if enable_transform:
+        if hasattr(prior.support, "base_constraint") and hasattr(
```

Comment:
we could also use `torch.distributions.biject_to(prior.support)` here to automatically find the right transform, no?

Reply:
I would vote for calling the transform variable `transform` instead of `tf`. 
And I think we could replace the first if-case here with
```python
transform = biject_to(prior.support)
```
and then construct the z-score transform by hand as it is done now, and then just before the return add
```python
if not isinstance(transform, IndependentTransform):
     transform = IndependentTransform(transform, reinterpredeted_batch_ndims=1)
```
This will make sure that calling `log_abs_det_jacobian` on the transform will sum over the parameter dimensions and will always return the batch-shape.